src/Utilities/ToolTask.cs

```diff
@@ -134,10 +134,30 @@ protected ToolTask(ResourceManager taskResources, string helpKeywordPrefix)
         /// </summary>
         public bool EchoOff { get; set; }
 
+        /// <summary>
+        /// This is the backing field for property <see cref="TaskProcessTerminationTimeout" />.
+        /// </summary>
+        private int _taskProcessTerminationTimeout;
+
         /// <summary>
         /// A timeout to wait for a task to terminate before killing it.  In milliseconds.
         /// </summary>
-        protected int TaskProcessTerminationTimeout { get; set; }
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the property is set to a negative value other than -1.</exception>
+        protected int TaskProcessTerminationTimeout
+        {
+            get => _taskProcessTerminationTimeout;
+            set
+            {
+                if (value < -1)
```

Comment:
```suggestion
                if (value < -1 && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_6))
```

Reply:
Could also simplify a bit with VerifyThrow...

But that should be equivalent from the user's perspective.

Reply:
@gpwen sorry for not giving precise guidance and then missing this PR.
`ValidateParameters()` is the best location for this validation - it'll ensure clean comunication of the issue to the user. Ideally error is logged there and `false` returned.

For simplicity - if change wave is not on (it is on by default - opting-out means somebody willingly wants legacy behavior) - I'd just set the value to the legacy one (`5000`) - regardless of validity of the value.

Other than that - it looks good.

Reply:
Implemented